### ISSUE TITLE
Add functionality to adjust project_role without managing the whole agency

### ISSUE THREAD

**Issue by victorgetz** [opentelekomcloud/terraform-provider-opentelekomcloud#1742]
There is a common problem if you create a project and then you want to use the cce you need to press this button which creates the project role for the "cce_admin_trust" agency.

You can currently create the agency with terraform but the problem is you need to configure it globaly if you are working in a subproject you have no chance to adjust just the project_role of "cce_admin_trust"

This destroys completly the automated approach and the indepent setup between projects.

My proposal would be the following.
Add a new resource:

```hcl

resource "opentelekomcloud_identity_agency_project_role_v3" "cce_allow" {
  agency_name  = "cce_admin_trust"
  project = "eu-de_subproject"
  roles = "Tenant Administrator"
}

```

With this resource you manage not the agency but the project entries.
That would help a lot

**Comment by aazon** [comment]
I don't see any business value (please correct me if I'm wrong) in creating a new CCE and not granting these "cce_admin_trust" permissions. Thus, it would make sense to perform this step automatically under the hood when Terraform creates a new CCE instance

**Comment by philippeckel** [comment]
I also do not see the need for a new resource. We're adding multiple project roles - `eu-de` & `MOS` & each sub-project - and this is working as intended:

```
resource "opentelekomcloud_identity_agency_v3" "cce_admin_trust" {
  name                  = "cce_admin_trust"
  description           = "Create by CCE Team"
  delegated_domain_name = "op_svc_cce"
  dynamic "project_role" {
    for_each = var.projects
    content {
      project = "eu-de_${project_role.key}"
      roles = [
        "Tenant Administrator"
      ]
    }
  }
  project_role {
    project = "eu-de"
    roles = [
      "Tenant Administrator"
    ]
  }
  project_role {
    project = "MOS"
    roles = [
      "Tenant Administrator"
    ]
  }
}
```

**Comment by aazon** [comment]
@philippeckel, this is exactly what @victorgetz wanted to avoid: to start working with a CCE in the arbitrary project, one needs to make changes in the root project and, therefore, have access there.

**Comment by victorgetz** [comment]
> I don't see any business value (please correct me if I'm wrong) in creating a new CCE and not granting these "cce_admin_trust" permissions. Thus, it would make sense to perform this step automatically under the hood when Terraform creates a new CCE instance

Yeah you are right Alex. This is a way much better way to solve the problem !

**Comment by victorgetz** [comment]
Any news?

**Comment by artem-lifshits** [comment]
Hi @victorgetz the idea of making additional resource (which in current config is intended to be used anyway) under the hood looks great at first but can lead to misunderstanding and remnant of orphan resources. Initially Terraform was meant to be declarative and explicit but creation of separate resource inside another breaks these standards.
Also take into account that to perform such automated resource creation CCE and IAM will have to use different client scope.

One of the ways how to use agency for CCE was described by @philippeckel. I've also made some research and found out that it's possible to create an agencies with permissions to all current and future projects. Sadly this feature is not available on API level and I've opened a demand for it: https://jira.tsi-dev.otc-service.com/browse/DM-4602

Hope this helps.
